src/core/executor.c

```diff
@@ -199,14 +199,15 @@ int main(int argc, char *argv[]) {
         log_set_prohibit_ipc(true);
         log_setup();
 
-        r = fdset_new_fill(/* filter_cloexec= */ 0, &fdset);
-        if (r < 0)
-                return log_error_errno(r, "Failed to create fd set: %m");
-
         r = parse_argv(argc, argv);
         if (r <= 0)
                 return r;
 
+        /* The serialization fd is set to CLOEXEC in parse_argv, so it's also filtered. */
+        r = fdset_new_fill(/* filter_cloexec= */ 0, &fdset);
```

Comment:
if you move this below the log setup while at it, we won't risk stuff going to the console

Reply:
I'll wait for #30249 to avoid merge conflict :)

Reply:
merged